### PR TITLE
add-context

### PR DESCRIPTION
**Pull request by aviadch**

### CODE REVIEW

File: frontend/src/MyContext.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+//import PHASE from "./App";
+//import WaitingRoom from "./WaitingRoom";
+//import WelcomeMenu from "./WelcomeMenu";
+export const MyContext = React.createContext();
+
+export const PHASE = {
+  START: 0,
+  WAITING_ROOM: 1,
```
Comment (AdiYElisha) [inline frontend/src/MyContext.js]: waiting room should be 0

File: frontend/src/MyContext.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  state = {
+    phase: PHASE.START
+  };
+
+  render() {
+    return (
+      <MyContext.Provider
+        value={{
+          state: this.state,
+          onNewGame: () => {
```
Comment (AdiYElisha) [inline frontend/src/MyContext.js]: create the function outside the render function, and pass it.
cleaner

File: frontend/src/MyContext.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
```
Comment (AdiYElisha) [inline frontend/src/MyContext.js]: file name is MyContext but it exports MyProvider. confusing

File: frontend/src/App.js [hunk 4]
```diff
@@ -2,28 +2,16 @@ import React, { Component } from "react";
...
-};
-class App extends Component {
-  constructor() {
-    super();
-    this.state = {
-      phase: PHASE.START
-    };
-  }
+import { MyContext } from "./MyContext";
+import MyProvider from "./MyContext.js";
```
Comment (AdiYElisha) [inline frontend/src/App.js]: these 2 lines make my confuse

File: frontend/src/App.js [hunk 5]
```diff
@@ -2,28 +2,16 @@ import React, { Component } from "react";
...
+import { PHASE } from "./MyContext.js";
 
-  onNewGame = () => {
-    this.setState({ phase: PHASE.WAITING_ROOM });
-  };
-
-  screenChoser = () => {
+class App extends Component {
+  componentChoser = phase => {
     let retComp = null;
```
Comment (AdiYElisha) [inline frontend/src/App.js]: retComp is very bad naming

File: frontend/src/App.js [hunk 6]
```diff
@@ -37,8 +25,18 @@ class App extends Component {
...
 
   render() {
-    const curComp = this.screenChoser();
-    return <div className="App">{curComp}</div>;
+    return (
+      <MyProvider>
+        <div className="App">
+          <MyContext.Consumer>
+            {context => {
+              const curComp = this.componentChoser(context.state.phase);
```
Comment (AdiYElisha) [inline frontend/src/App.js]: naming

File: frontend/src/WelcomeScreen.js [hunk 7]
```diff
@@ -10,19 +11,29 @@ class WelcomeScreen extends Component {
...
+                <div>
+                  <Button
+                    onClick={context.onNewGame}
+                    variant="contained"
+                    color="primary"
+                  >
+                    Start a New Game
+                  </Button>
+                </div>
+
```
Comment (AdiYElisha) [inline frontend/src/WelcomeScreen.js]: whitespace

File: frontend/src/WelcomeScreen.js [hunk 8]
```diff
@@ -10,19 +11,29 @@ class WelcomeScreen extends Component {
...
+                <div>
+                  <Button variant="contained" color="primary">
+                    Join an Existing Game by ID
```
Comment (AdiYElisha) [inline frontend/src/WelcomeScreen.js]: "Join Game"

File: frontend/src/MyContext.js (hunk 1 above)
Comment (aviadch) [reply frontend/src/MyContext.js]: Sure? I changed the 0 key to be WELCOME_SCREEN from start as it should be. I think it is now ok

File: frontend/src/MyContext.js (hunk 3 above)
Comment (aviadch) [reply frontend/src/MyContext.js]: I will try to explain:
The convention I saw in the tutorial do exactly that - Provider And Context in the same file.
I have split the classes to two separate files - MyProvider.js and MyContext.js. look how small is MyContext.js now:

```
import React from "react";

export const MyContext = React.createContext();
```
that's it. isn't it too small for a separate file?

File: frontend/src/App.js (hunk 4 above)
Comment (aviadch) [reply frontend/src/App.js]: look at the comment above

File: frontend/src/App.js (hunk 5 above)
Comment (aviadch) [reply frontend/src/App.js]: done

File: frontend/src/App.js (hunk 6 above)
Comment (aviadch) [reply frontend/src/App.js]: done

File: frontend/src/MyContext.js (hunk 3 above)
Comment (AdiYElisha) [reply frontend/src/MyContext.js]: right, but the convention is that the file name is MyContext and not MyProvider? if so, let it be

Comment (aviadch) [reply frontend/src/MyContext.js]: Yes and the reason is that you are using context from a provider or a consumer and for that you need this import:
`import { MyContext } from "./MyContext";`

for now i left it that way. I think using separate files will make it more clearer

**Pull request merged**